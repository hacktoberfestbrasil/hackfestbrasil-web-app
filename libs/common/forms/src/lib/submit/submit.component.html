<ng-template #content><ng-content></ng-content></ng-template>

<button *ngIf="!isButton; else buttonTmpl" type="submit">
  <ng-container *ngTemplateOutlet="content"></ng-container>
  <svg viewBox="0 0 320 140">
    <path d="M10,10 l300,0 0,120 -300,0 0,-120"></path>
  </svg>
</button>

<ng-template #buttonTmpl>
  <ng-container *ngTemplateOutlet="content"></ng-container>
  <svg viewBox="0 0 320 140">
    <path d="M10,10 l300,0 0,120 -300,0 0,-120"></path>
  </svg>
</ng-template>

