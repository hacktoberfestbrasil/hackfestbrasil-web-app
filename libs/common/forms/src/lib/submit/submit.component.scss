:host {
  --hackfest-button-width: 300px;
  --hackfest-button-height: 120px;
  --hackfest-button-font-size: 32px;

  display: inline-flex;
  align-items: center;
  justify-content: center;

  &,
  button {
    width: var(--hackfest-button-width);
    height: var(--hackfest-button-height);
    border: none;
    color: #ffffff50;
    background-color: #93c2db25;
    cursor: pointer;
    padding: 0;
    position: relative;
    line-height: var(--hackfest-button-height);
    font-size: var(--hackfest-button-font-size);
    font-weight: bold;
    transition: color 400ms ease-in-out, background-color 600ms ease 400ms;
  }

  & svg,
  button svg {
    position: absolute;
    top: -10px;
    left: -10px;
    width: calc(var(--hackfest-button-width) + 20px);
    height: calc(var(--hackfest-button-height) + 20px);
    fill: none;
    stroke: #fff;
    stroke-linecap: round;
    stroke-dasharray: 840;
    stroke-dashoffset: 840;
    stroke-width: 5;
    transition: 600ms stroke-dashoffset ease;
  }

  &:hover,
  button:hover {
    color: rgba(255, 138, 226, 1);
    background-color: #93c2db50;
    svg {
      stroke-dashoffset: 0;
    }
  }
}
