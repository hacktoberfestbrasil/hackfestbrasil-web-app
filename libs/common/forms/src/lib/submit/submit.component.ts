import { Component, ElementRef, Input } from '@angular/core';

@Component({
  selector: `
    hackfest-submit,
    button[hackfest-submit]
  `,
  templateUrl: './submit.component.html',
  styleUrls: ['./submit.component.scss'],
})
export class HackfestSubmitComponent {
  @Input()
  public set disabled(value: boolean) {
    this._disabled = value;
  }
  private _disabled: boolean;
  public get disabled(): boolean {
    return this._disabled;
  }

  get el() {
    return this._el.nativeElement;
  }
  get isButton() {
    return this.el?.tagName === 'BUTTON';
  }

  constructor(private _el: ElementRef<HTMLElement>) {}
}
