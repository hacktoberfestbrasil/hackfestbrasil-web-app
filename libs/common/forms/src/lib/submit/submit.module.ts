import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HackfestSubmitComponent } from './submit.component';

export * from './submit.component';
export * from './submit.module';

@NgModule({
  declarations: [HackfestSubmitComponent],
  imports: [CommonModule],
  exports: [HackfestSubmitComponent],
})
export class HackfestSubmitModule {}
