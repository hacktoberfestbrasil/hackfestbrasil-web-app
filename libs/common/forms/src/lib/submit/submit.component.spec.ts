import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest';
import { HackfestSubmitComponent } from './submit.component';

const VIEWS = {
  element: `<hackfest-submit></hackfest-submit>`,
  button: `<button hackfest-submit>Submit</button>`,
  buttonDisabled: `<button disabled="true" hackfest-submit>Submit</button>`,

}

describe('HackfestSubmitComponent', () => {
  let spectator: SpectatorHost<HackfestSubmitComponent>;

  const createHost = createHostFactory(HackfestSubmitComponent);


  describe('Views', () => {
    it('should create snapshot', () => {
      spectator = createHost(VIEWS.element);
      expect(spectator.component).toMatchSnapshot();
    });

    it('should create button snapshot', () => {
      spectator = createHost(VIEWS.button);
      expect(spectator.component).toMatchSnapshot();
    });

    it('should create button disabled', () => {
      spectator = createHost(VIEWS.buttonDisabled);
      expect(spectator.component).toMatchSnapshot();
    });
  })

  it('should create', () => {
    spectator = createHost(`<hackfest-submit></hackfest-submit>`);
    expect(spectator.component).toBeTruthy();
  });

  it('should have submit text', () => {
    spectator = createHost(`<hackfest-submit>Submit</hackfest-submit>`);
    expect(spectator.query('button')).toHaveText('Submit');
  });

  it('should have text with button as host', () => {
    spectator = createHost(`<button hackfest-submit>Submit</button>`);
    expect(spectator.element).toHaveText('Submit');
  });

  it('should be disabled', () => {
    spectator = createHost(`<button disabled="true" hackfest-submit>Submit</button>`);
    expect(spectator.element.hasAttribute('disabled')).toBeTruthy();
  });

  it('should have svg and button elements', () => {
    spectator = createHost(`<hackfest-submit>Bot√£o</hackfest-submit>`);
    expect(spectator.query('svg')).toExist();
    expect(spectator.query('button')).toExist();
  });

});
