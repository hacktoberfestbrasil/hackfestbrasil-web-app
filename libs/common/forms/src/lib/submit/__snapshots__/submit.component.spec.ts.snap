// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HackfestSubmitComponent Views should create button disabled 1`] = `
HackfestSubmitComponent {
  "_disabled": "true",
  "_el": ElementRef {
    "nativeElement": <button
      disabled="true"
      hackfest-submit=""
    >
      
      
      
      Submit
      
      <svg
        viewBox="0 0 320 140"
      >
        <path
          d="M10,10 l300,0 0,120 -300,0 0,-120"
        />
      </svg>
      
    </button>,
  },
}
`;

exports[`HackfestSubmitComponent Views should create button snapshot 1`] = `
HackfestSubmitComponent {
  "_el": ElementRef {
    "nativeElement": <button
      hackfest-submit=""
    >
      
      
      
      Submit
      
      <svg
        viewBox="0 0 320 140"
      >
        <path
          d="M10,10 l300,0 0,120 -300,0 0,-120"
        />
      </svg>
      
    </button>,
  },
}
`;

exports[`HackfestSubmitComponent Views should create snapshot 1`] = `
HackfestSubmitComponent {
  "_el": ElementRef {
    "nativeElement": <hackfest-submit>
      
      
      <button
        type="submit"
      >
        
        
        <svg
          viewBox="0 0 320 140"
        >
          <path
            d="M10,10 l300,0 0,120 -300,0 0,-120"
          />
        </svg>
      </button>
      
    </hackfest-submit>,
  },
}
`;
